### PR TITLE
ENCM-115 Add functional characterization matrix

### PR DESCRIPTION
**Pull request by forresttanaka**
* The main addition exists in the matrix_functional_characterization.js file which mostly follows the pattern of the simpler matrix-rendering files.
* I changed `<MatrixAddCart>` to only request the `@id` and `@type` of the related datasets to reduce the size of the results, in some cases very radically. Those two properties are the only ones that the cart code uses.
* I changed `<SearchFilter>` to allow matrix pages to optionally pass the specific title they want for the search-box label.

### CODE REVIEW

**Comment by forresttanaka** [comment]
@emro Ugh. I left an old URL in the menu. It should be identical to the URL in the FC home-page button. I’ll push an update to that momentarily. That’s the only change.

**Review by emro — DISMISSED** [review]
Looks awesome! I like the colors and it works well.

File: src/encoded/search_views.py [hunk 1]
```diff
@@ -38,6 +38,7 @@
 from snosearch.fields import AllResponseField
 from snosearch.fields import BasicMatrixWithFacetsResponseField
 from snosearch.fields import MissingMatrixWithFacetsResponseField
+from snosearch.fields import MultitypeMissingMatrixWithFacetsResponseField
```
Comment (emro) [inline src/encoded/search_views.py:41]: This is almost not worth noting but I might have done the cases like this: MultiTypeMissingMatrixWithFacetsResponseField

**Review by emro — APPROVED** [review]
Looks great!

**Pull request merged**